Simplified/Audiobooks/Bookmark/NYPLAudiobookBookmarksBusinessLogic.swift

```diff
@@ -9,11 +9,23 @@
 import Foundation
 import NYPLAudiobookToolkit
 
-class NYPLAudiobookBookmarksBusinessLogic: NYPLAudiobookBookmarksBusinessLogicDelegate {
+class NYPLAudiobookBookmarksBusinessLogic: NYPLAudiobookBookmarking {
   
   // MARK: - Properties
   
-  var bookmarks: [NYPLAudiobookBookmark]
+  var bookmarks: [NYPLAudiobookBookmark] {
+    didSet {
+      updateBookmarksDictionary()
+    }
+  }
+  
+  /// Given that we need to constantly check if there exists a bookmark
```

Comment:
@ettore Feel free to make any suggestions if you have a better solution to improve the performance.